nixos/modules/services/network-filesystems/samba.nix

```diff
@@ -74,38 +74,22 @@ in
     # !!! clean up the descriptions.
 
     services.samba = {
+      enable = lib.mkEnableOption "Samba, the SMB/CIFS protocol";
 
-      enable = mkOption {
-        type = types.bool;
-        default = false;
-        description = ''
-          Whether to enable Samba, which provides file and print
-          services to Windows clients through the SMB/CIFS protocol.
-
-          ::: {.note}
-          If you use the firewall consider adding the following:
-
-              services.samba.openFirewall = true;
-          :::
-        '';
+      package = lib.mkPackageOption pkgs "samba" {
+        example = "samba4Full";
       };
 
-      openFirewall = mkOption {
-        type = types.bool;
-        default = false;
-        description = ''
-          Whether to automatically open the necessary ports in the firewall.
-        '';
-      };
+      openFirewall = lib.mkEnableOption "automatically open the necessary ports in the firewall.";
```

Comment:
That would now be completed to ``Wether to enable automatically open the necessary ports in the firewall..``.